airflow/jobs/scheduler_job.py

```diff
@@ -1899,3 +1910,17 @@ def adopt_or_reset_orphaned_tasks(self, session: Session = None):
                     raise
 
         return len(to_reset)
+
+    @provide_session
+    def check_trigger_timeouts(self, session: Session = None):
+        """
+        Looks at all tasks that are in the "deferred" state and whose trigger
+        timeout has passed, so they can be marked as failed.
+        """
+        timed_out_tasks = session.query(TaskInstance).filter(
+            TaskInstance.state == State.DEFERRED, TaskInstance.trigger_timeout < timezone.utcnow()
+        )
+        num_tasks = timed_out_tasks.count()
```

Comment:
Should we change this to use `func.count` ? We will have an extra line or two and some duplication but more reliable:

https://docs.sqlalchemy.org/en/14/orm/query.html#sqlalchemy.orm.Query.count
https://stackoverflow.com/questions/14754994/why-is-sqlalchemy-count-much-slower-than-the-raw-query

Reply:
What do you mean by `func.count`? Do you mean limiting the count query to only be over a single column or something?

Reply:
Oh, I see! SQLAlchemy is less clever than I thought, then. I'll add it in.